Yank/repex.py

```diff
@@ -222,7 +222,22 @@ def _are_subfiles_open(self):
                 open_check_list.append(storage.isopen())
         return np.all(open_check_list)
 
-    def open(self, mode='r', convention='ReplicaExchange'):
+    def _create_dimension_if_needed(self, dimname, dimsize):
```

Comment:
Should this also check that the size of the dim is the size we want to shape it? I.e. raise an error or a warning if the `dimsize` argument does not match the `dimsize` of the exiting dimension?

Reply:
Great idea! I'll add this.

Reply:
@Lnaden : What kind of `Exception` should we raise if there is a mismatch?

Reply:
Looking at it some more, it may not be needed since this is an internal function (i.e. the user should never call it so we should know what we are doing as devs when we invoke it). That said, I would make it a `ValueError` by default, but if you wanted to make this a formal assert, it could be an `AssertionError`

Reply:
I'll use `ValueError`.

Reply:
Addressed!